build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -212,16 +216,26 @@ class WatchImpl implements BuildState {
         })
         .asyncMap<AssetChange>((change) {
           // Kill future builds if the root packages file changes.
+          var id = change.id;
           assert(originalRootPackagesDigest != null);
-          if (change.id != rootPackagesId) return change;
-          return environment.reader.readAsBytes(rootPackagesId).then((bytes) {
-            if (md5.convert(bytes) != originalRootPackagesDigest) {
-              _terminateCompleter.complete();
-              _logger.severe('Terminating builds due to package graph update, '
-                  'please restart the build.');
-            }
-            return change;
-          });
+          if (id == rootPackagesId) {
+            return environment.reader.readAsBytes(rootPackagesId).then((bytes) {
+              if (md5.convert(bytes) != originalRootPackagesDigest) {
+                _terminateCompleter.complete();
+                _logger
+                    .severe('Terminating builds due to package graph update, '
+                        'please restart the build.');
+              }
+              return change;
+            });
+          } else if (p
+              .basename(id.path)
+              .contains(new RegExp('^build(\.$_configKey)?\.yaml\$'))) {
```

Comment:
I think we also need to terminate for all `build.yaml` file changes. That file impacts the build script.

Reply:
Yes, this catches both `build.yaml` and `build.<config>.yaml`